src/search/task.cc

```diff
@@ -36,22 +36,22 @@ void Task::create_empty_initial_state(size_t number_predicates)
     }
     initial_state = DBState(move(fluents), vector<bool>(predicates.size(), false));
     static_info = StaticInformation(move(static_preds), vector<bool>(predicates.size(), false));
-    initial_state.nullary_atoms.clear();
-    initial_state.nullary_atoms.resize(number_predicates, false);
 }
 
 void Task::dump_state(DBState s) const
 {
     /*
      * Output initial state in a human readable way.
      */
-    for (size_t j = 0; j < s.nullary_atoms.size(); ++j) {
-        if (s.nullary_atoms[j])
+    auto nullary_atoms = s.get_nullary_atoms();
+    for (size_t j = 0; j < nullary_atoms.size(); ++j) {
+        if (nullary_atoms[j])
             cout << predicates[j].getName() << ", ";
     }
-    for (size_t i = 0; i < s.relations.size(); ++i) {
+    auto relations = s.get_relations();
```

Comment:
I prefer `const auto&` here - honestly don't remember if it makes a difference according to the spec, but I find it clearer...